source/utilities.tex

```diff
@@ -12088,7 +12088,7 @@
 \remarks
 Two \tcode{weak_ptr} objects are equivalent if
 they store the same pointer value and
-either share ownership, or both are empty.
+share ownership, or if both are empty.
```

Comment:
This change, however, I believe is valid: there's no way to construct a `weak_ptr` that both stores a pointer value and is empty.

Reply:
Doesn't this do so?

    static int i = 0;
    std::shared_ptr<int> empty;
    std::shared_ptr<int> empty_not_null(empty, &i);
    std::weak_ptr<int> weak(empty_not_null);

And you can get another empty-not-null `shared_ptr` by constructing it from `weak` (although not by calling `weak.lock()` because that returns `shared_ptr<int>()` if it's empty.

Reply:
> Doesn't this do so?

I initially thought so as well, but [util.smartptr.weak.const]/4 says:

> *Effects:* If `r` is empty, constructs an empty `weak_ptr` object; otherwise, constructs a `weak_ptr` object
that shares ownership with `r` and stores a copy of the pointer stored in `r`.

and now I'm wondering if I implement that correctly.

Reply:
> and now I'm wondering if I implement that correctly.

Nope: we get this wrong.

Reply:
I checked the ctor efects, but apparently I can't read.

> Nope: we get this wrong.

￼I think we do too.

Reply:
Even if you can't create such a weak ptr, is using subtly different wording here an improvement?

Reply:
> Even if you can't create such a `weak ptr`, is using subtly different wording here an improvement?

Probably not, and the existing wording will be correct if/when we apply the resolution of the LWG issue I just submitted. It may be better to leave this wording alone.